### ISSUE TITLE
[docs]: How does remote development work now?

### ISSUE THREAD

**Issue by ian-buse** [buehler/dotnet-operator-sdk#742]
### Describe the missing piece of documentation

Prior to v 8.x, there was good documentation on how to get your operator up and running, how to create and install webhooks, etc. Running the 'install' and then 'webhooks install' command worked well. I skipped 8.x.

With 9.0.0, there seems to be missing or out of date documentation for developing the operator, _especially_ in regard to webhooks. The webhook commands are missing, and the [CLI documentation](https://github.com/buehler/dotnet-operator-sdk/blob/main/src/KubeOps.Cli/README.md) references commands that as far as I can tell no longer exist.

In a somewhat related problem, the webhook example [`Program.cs`](https://github.com/buehler/dotnet-operator-sdk/blob/main/examples/WebhookOperator/Program.cs) seems to reference a `RegisterComponents()` method that no longer exists, with the error `'IOperatorBuilder' does not contain a definition for 'RegisterComponents' and no accessible extension method 'RegisterComponents' accepting a first argument of type 'IOperatorBuilder' could be found (are you missing a using directive or an assembly reference?)`

Part of these issues were brought up in #740, but I wanted to make sure this gets seen. I have not been unable to figure out how to get things working.

**Comment by SGStino** [comment]
If i'm correct, the RegisterComponents was doing assembly scanning to find all the required components.
 
 Now you can register all components explicitly:
 ```csharp
services.AddKubernetesOperator()
  .AddController<MyController, MyControlledResource>()
  .AddFinalizer<MyFinalizer, MyControlledResource>("my-finalizer-identifier");
 ```

**Comment by buehler** [comment]
Hey @ian-buse 

If you do not use source generators (`KubeOps.Generator`), then you do not have the RegisterComponents method. However, if the generator is in place, it will create the required method for "register components" and as such, all things are getting registered.

You can see the code that is generated in the test files (https://github.com/buehler/dotnet-operator-sdk/tree/main/test/KubeOps.Generator.Test).

The webhook install command and other commands regarding webhooks are no longer required. When creating the kustomize configs with the kubeops.cli, the cli detects if you have webhooks installed. If you do, it creates a self signed certificate (now without the help of the external tool) and the required resources.

https://buehler.github.io/dotnet-operator-sdk/src/KubeOps.Operator.Web/README.html#installing-in-the-cluster

**Comment by SGStino** [comment]
No, the help is talking about much more than just these, which might confuse people :)

**Comment by ian-buse** [comment]
I see now how this works now, thank you. I've opted not to use RegisterComponents() and do the registration manually.

This still leaves a small problem with the documentation because it talks about a few commands that don't exist anymore. I'd be happy to remove the old commands from the documentation if you're okay with that.
